xonsh/environ.py

```diff
@@ -942,6 +943,20 @@ def DEFAULT_VARS():
             "Note: ``erasedups`` is supported only in sqlite backend).",
             doc_store_as_str=True,
         ),
+        "HOSTNAME": Var(
+            is_string,
+            ensure_string,
+            ensure_string,
+            uname().node,
```

Comment:
This should be something like: `default_value(lambda env: uname.node)` so it is not called when the deafults are created.

Reply:
I actually wanted it to be called only once.

Reply:
Done.

Reply:
Done.